fsutils/afero.go

```diff
@@ -0,0 +1,42 @@
+package fsutils
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+func SetupTemporaryFiles(fs afero.Fs) (file afero.File, dir string, err error) {
+	tmpf, err := afero.TempFile(fs, "", "tar-file-")
+	if err != nil {
+		return nil, "", err
+	}
+
+	tmpd, err := afero.TempDir(fs, "", "tar-dir-")
+	if err != nil {
+		return nil, "", err
+	}
+	return tmpf, tmpd, err
+}
+
+func GetRepoFolder(fs afero.Fs, tmpd string) (string, error) {
+	files, err := afero.ReadDir(fs, tmpd)
+	if err != nil {
+		return "", err
+	}
+	if len(files) != 1 {
+		return "", fmt.Errorf("expected only one folder from archive tar, found (%d)", len(files))
```

Comment:
similar comment to above, this util has some strong assumptions, consider privatizing it or renaming

Reply:
fixed